packages/docusaurus-plugin-sitemap/src/createSitemap.ts

```diff
@@ -16,17 +16,19 @@ export default function createSitemap(
 ) {
   const {url: hostname} = siteConfig;
   if (!hostname) {
-    throw new Error('Url in docusaurus.config.js cannot be empty/undefined');
+    throw new Error('url in docusaurus.config.js cannot be empty/undefined');
   }
 
-  const urls = routesPaths.map(
-    (routesPath) =>
-      ({
-        url: routesPath,
-        changefreq: options.changefreq,
-        priority: options.priority,
-      } as SitemapItemOptions),
-  );
+  const urls = routesPaths
+    .filter((route: string) => route !== '/404.html')
```

Comment:
Could it be possible to create from this new constant in the according file? https://github.com/facebook/docusaurus/blob/master/packages/docusaurus/src/constants.ts

https://github.com/facebook/docusaurus/blob/1f00d15c7439fdbfb37d57f0cafdaedb2b34ef41/packages/docusaurus/src/server/routes.ts#L50

Reply:
We already have a constant named routesPaths

Reply:
I meant the value "/404.html" put in a constant.

Reply:
I don't really see a point, its already a static string literal

Reply:
Alexey means to reduce duplication of the string `'404.html` so if we ever change it in `src/server/routes.ts`, we don't have to change it here as well. I think the current approach is fine too as the test will fail if we change it in `src/server/routes.ts` but not here.

Reply:
@RDIL I'm not sure if `route !== '/404.html'` will be true for a site a `baseUrl` which is not `/`. Can you test with a non-empty `baseUrl`? We will need to be smarter about the check and use `String.prototype.endsWith()` otherwise.

Reply:
Fixed.